lib/presentation/widgets/search_list.dart

```diff
@@ -0,0 +1,113 @@
+}
+
+class _SearchListState extends State<SearchList> {
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_onScroll);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final loading = BlocProvider.of<SearchBloc>(context).state.loading;
+    final characters = BlocProvider.of<SearchBloc>(context).state.charactersViewData;
+    final error = BlocProvider.of<SearchBloc>(context).state.error;
+    final noResult = BlocProvider.of<SearchBloc>(context).state.noResult;
+    final emptySearchField = BlocProvider.of<SearchBloc>(context).state.emptySearchField;
+    final hasReachedMax = BlocProvider.of<SearchBloc>(context).state.hasReachedMax;
+    final afterScroll = BlocProvider.of<SearchBloc>(context).state.afterScroll;
+
+    String nameStartsWith = widget.nameStartsWith;
+    return Container(child: () {
+      if (emptySearchField == true) {
+        return Container();
+      }
+      if (loading == true && afterScroll == false) {
+        return const Center(
+          child: CircularProgressIndicator(),
+        );
+      }
+      if (characters != null && noResult == false) {
+        return ListView.builder(
+          controller: _scrollController,
+          itemCount: hasReachedMax ? characters.length : characters.length + 1,
+          itemBuilder: (BuildContext context, int index) {
+            Widget endOfPage;
+            if (error != null) {
+              endOfPage = SearchBottomError(nameStartsWith: nameStartsWith);
+            } else {
+              endOfPage = const BottomLoader();
+            }
+            return index == characters.length
+                ? endOfPage
+                : SizedBox(
+                    height: 95,
+                    child: GestureDetector(
+                      onTap: () {
+                        Navigator.pushNamed(context, AppRoutes.detailPage, arguments: characters[index].id);
+                      },
+                      child: CharacterCardWidget(characters[index]),
+                    ),
+                  );
+          },
+        );
+      }
+      if (error != null) {
+        return ErrorPage(
+          onRetry: () {
+            BlocProvider.of<SearchBloc>(context).add(SearchedCharacterEvent(nameStartsWith: nameStartsWith));
+          },
+          error: error.toString(),
+        );
+      }
+      if (noResult == true) {
+        return const Center(
+          child: Text(
+            'No Results',
+            style: TextStyle(fontSize: 25),
+          ),
+        );
+      }
+      return const SizedBox.shrink();
+    }());
+  }
+
+  @override
+  void dispose() {
+    _scrollController
+      ..removeListener(_onScroll)
+      ..dispose();
+    super.dispose();
+  }
+
+  void _onScroll() {
+    if (_isBottom) context.read<SearchBloc>().add(ScrolledToEndSearchEvent(nameStartsWith: widget.nameStartsWith));
```

Comment:
You should use callbacks (lambdas) to interact with the parent view, do not use Bloc in your widgets.

Reply:
done